### DISCUSSION TITLE
Some battery powered end devices leaving the network after a restart

### DISCUSSION THREAD

**Discussion by donutVillan** [Koenkk/zigbee2mqtt#16358] [General]
I restarted the computer and after that several battery powered devices are missing from the UI devices list.
They are also missing when creating the map.
Nothing in the log when pressing the buttons.

They are, however, still in the configuration.yaml and state.json file.

Any ideas?

**Comment by donutVillan** [comment]
And the plot thickens...

It's only Hue wall switches that are gone. But not all of them. Only some.

I added another one and then restarted the computer and that one also disappeared.
The other once binded to the controller through routers but this one was connected directly to the controller (not that I think it matters for this problem).

In the log there is the following message.
`warn  2023-01-26 05:53:40: Device 'Home Office Wall Switch' left the network`

And like the other once it's still in the configuration.yaml and state.json file.

**Reply by donutVillan** [reply]
I changed the channel to 25 before creating the network and I've seen some posts where the channel had reverted for some people but it looks like it's still running on 25.

After a restart the log shows:
`debug 2023-01-26 09:04:37: Zigbee network parameters: {"channel":25,"extendedPanID":221,"panID":56460}`

**Comment by donutVillan** [comment]
This [https://github.com/Koenkk/zigbee2mqtt/issues/16343](https://github.com/Koenkk/zigbee2mqtt/issues/16343) could be connected.

Both of us are running a EZSP controller.
I have the Sonoff ZBDongle-E.

Does someone else have problems with battery powered devices after a restart?

**Reply by ctrld-pl** [reply]
I had the same issues with ZBDongle-E
Especially with [Xiaomi SRTS-A01](https://www.zigbee2mqtt.io/devices/SRTS-A01.html#xiaomi-srts-a01) and [SONOFF SNZB-02](https://www.zigbee2mqtt.io/devices/SNZB-02.html)

**Comment by donutVillan** [comment]
I now reset the Hue wall switch that previously left the network.
At this point the device was still in the configuration.yaml but no longer in the state.json file.

`warning 2023-01-26 18:36:19 Device 'Home Office Wall Switch' left the network`

Lo and behold, now the switch left for real...

Now I could add it again. It even adds with the correct friendly name (which I assume is because it is still in the configuration.yaml file)

And a restart directly results in 
`warn  2023-01-26 18:51:41: Device 'Home Office Wall Switch' left the network`

**Comment by donutVillan** [comment]
I moved over to the Sonoff ZBDongle-P and that seems to have "solved" the problem.

As a side note I should mention that when I started deleting the other devices from my network the previously disconnected wall switches started coming back. That was nice since I didn't have to pull them out of wall to reset them then.

**Comment by pauloon** [comment]
I still have this same issue very randomly (not only after restarts), more frequent with some battery devices like Sonoff Door Sensors, but it already happened with other devices too. This weekend I left off some powered LED strips that use Lonsonho router controllers for around 24 hours and when I turned them on they didn't work. Only returned after repairing.
What could be this?
@Koenkk , did you notice this?
Please, help.

Thanks,
Paulo.

**Reply by donutVillan** [reply]
What do you mean with "still"? Did you also switch from a EZSP controller?

**Reply by pauloon** [reply]
I still have this issue of battery devices leaving network or losing pairing after a restart. Not sure what you mean with "EZSP"....

**Reply by donutVillan** [reply]
which controller do you have?

**Reply by pauloon** [reply]
I'm using a Sonoff USB dongle-P.
